core/src/main/scala/sttp/openai/requests/embeddings/EmbeddingsResponseBody.scala

```diff
@@ -0,0 +1,33 @@
+package sttp.openai.requests.embeddings
+
+import sttp.openai.json.SnakePickle
+
+object EmbeddingsResponseBody {
+  case class EmbeddingData(
+      `object`: String,
+      index: Int,
+      embedding: Seq[Double]
+  )
+  object EmbeddingData {
+    implicit val embeddingDataReader: SnakePickle.Reader[EmbeddingData] = SnakePickle.macroR[EmbeddingData]
```

Comment:
[minor] Maybe it would be enough to call `SnakePickle.macroR` (no type parameter - it's inferred)
[question] is `macroR` a upickle-defined method? the name is the quite poor ;) (it should be `derived` or sth like that, it's not interesting that it's a macro)

Reply:
`[minor] Maybe it would be enough to call SnakePickle.macroR (no type parameter - it's inferred)`
I guess not, compiler complains:
```
[error] 12 |    implicit val embeddingDataReader = SnakePickle.macroR[EmbeddingData]
[error]    |                 ^
[error]    |                type of implicit definition needs to be given explicitly
```
`[question] is macroR a upickle-defined method? the name is the quite poor ;) (it should be derived or sth like that, it's not interesting that it's a macro)`
Yes, it is upickle-defined method :)

Reply:
> I guess not, compiler complains:

Yeah the type annotation must remain, I meant the type parameter to `macroR`

Reply:
Oh, you are right. Got rid of them in `embeddings` module for now and will do a clean-up in others in next PR